test-runner-poc/environments.js

```diff
@@ -75,6 +71,15 @@ class ReplicaSetEnvironment extends EnvironmentBase {
     }
   }
 }
+
+function generateNodesArray(hosts, configFunc){
+  let array = [];
+  for (let i = 0; i < hosts.length; i++) {
+    array[i] = configFunc(hosts[i].port || hosts[0].port + i);
```

Comment:
It looks like `shardOptions` are going away when we're switching to the use of `configFunc`. Is it important to keep the `shardOptions`?

Reply:
added `shardOptions` back in.

Reply:
Where are `shardOptions`?

Reply:
They are in `genShardedConfig`, but the line that sets a nodes array to the return value of `genShardedConfig` (on line 145) is also commented out.